spec/java_buildpack/framework/app_dynamics_agent_spec.rb

```diff
@@ -165,6 +165,18 @@
           component.compile
         end
       end
+
+      context do
+
+        let(:environment) { { 'APPD_CONF_DIR' => 'BOOT-INF/classes/appdynamics/conf' } }
```

Comment:
I am not sure this test is really testing anything. For example, if I change this path to `BOOT-INF/classes/appdynamics/foo`, the test still passes. This is happening because the `stub-app-dynamics-agent.zip` fixture does not have a version directory.

To ensure this functionality works I think we need to add a directory like `ver1.2.3` to the text fixture and then assert that the provided `app-agent-config.xml` is copied into the version directory.

For example, if we changed the contents of `spec/fixtures/framework_app_dynamics_agent/BOOT-INF/classes/appdynamics/conf/app-agent-config.xml` to a simple test string we could add an assertion along the lines of
```
expect(IO.read(sandbox + "ver1.2.3"+"conf"+"app-agent-config.xml")).to eq("app-agent-config-from-dir")
```